wrapper/src/test/java/integration/refactored/container/tests/HikariTests.java

```diff
@@ -101,18 +129,179 @@ public void testOpenConnectionWithDataSourceClassName() throws SQLException {
     targetDataSourceProps.setProperty(PropertyDefinition.PLUGINS.name, "");
     // For MariaDB tests, MariaDbDataSource only accepts the url parameter.
     targetDataSourceProps.setProperty("url", ConnectionStringHelper.getUrl());
-    ds.addDataSourceProperty("targetDataSourceProperties", targetDataSourceProps);
+    dataSource.addDataSourceProperty("targetDataSourceProperties", targetDataSourceProps);
 
-    Connection conn = ds.getConnection();
+    final Connection conn = dataSource.getConnection();
 
     assertTrue(conn instanceof HikariProxyConnection);
-    HikariProxyConnection hikariConn = (HikariProxyConnection) conn;
+    final HikariProxyConnection hikariConn = (HikariProxyConnection) conn;
 
     assertTrue(hikariConn.isWrapperFor(ConnectionWrapper.class));
-    ConnectionWrapper connWrapper = (ConnectionWrapper) hikariConn.unwrap(Connection.class);
+    final ConnectionWrapper connWrapper = (ConnectionWrapper) hikariConn.unwrap(Connection.class);
     assertTrue(connWrapper.isWrapperFor(DriverHelper.getConnectionClass()));
 
     assertTrue(conn.isValid(10));
     conn.close();
   }
+
+  /**
+   * After getting successful connections from the pool, the cluster becomes unavailable.
+   */
+
+  @TestTemplate
+  @DisableOnTestDriver(TestDriver.MARIADB)
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @EnableOnTestFeature({TestEnvironmentFeatures.NETWORK_OUTAGES_ENABLED})
+  @EnableOnNumOfInstances(min = 3)
+  public void testFailoverLostConnection() throws SQLException {
+    final Properties customProps = new Properties();
+    customProps.setProperty(PropertyDefinition.PLUGINS.name, "failover");
+    customProps.setProperty("failoverTimeoutMs", Integer.toString(1));
+    DriverHelper.setSocketTimeout(customProps, 1, TimeUnit.SECONDS);
+
+    final HikariDataSource dataSource = createDataSource(customProps);
+
+    try (Connection conn = dataSource.getConnection()) {
+      assertTrue(conn.isValid(5));
+
+      ProxyHelper.disableAllConnectivity();
+
+      assertThrows(FailoverFailedSQLException.class, () -> auroraUtil.queryInstanceId(conn));
+      assertFalse(conn.isValid(5));
+    }
+
+    assertThrows(SQLTransientConnectionException.class, dataSource::getConnection);
+    ProxyHelper.enableAllConnectivity();
+  }
+
+  /**
+   * After getting a successful connection from the pool, the connected instance becomes unavailable and the
+   * connection fails over to another instance through the Enhanced Failure Monitor.
+   */
+  @TestTemplate
+  @EnableOnDatabaseEngineDeployment(DatabaseEngineDeployment.AURORA)
+  @EnableOnTestDriver({TestDriver.PG, TestDriver.MYSQL})
```

Comment:
I see Karen already asked a similar question that was resolved, so if you've already figured this out you can ignore this comment, but should we change this to @DisableOnTestDriver(TestDriver.MARIADB)? Or can we remove it altogether (does this test fail for mariadb? Why?)

Reply:
I've changed it, just need to push these changes. Same as above, looking into a fix